stdlib/2and3/decimal.pyi

```diff
@@ -49,24 +55,37 @@ class Overflow(Inexact, Rounded): ...
 
 class Underflow(Inexact, Rounded, Subnormal): ...
 
-def setcontext(context: Context): ...
+if sys.version_info >= (3,):
+    class FloatOperation(DecimalException, TypeError): ...
+
+def setcontext(context: Context) -> None: ...
 def getcontext() -> Context: ...
 def localcontext(ctx: Optional[Context] = ...) -> _ContextManager: ...
 
-class Decimal(SupportsAbs[Decimal], SupportsFloat, SupportsInt):
-    def __init__(cls, value: Union[_Decimal, float, str, unicode,
+if sys.version_info >= (3,):
+    _SupportsRound = SupportsRound[int]
+else:
+    _SupportsRound = object
+
+class Decimal(SupportsAbs[Decimal], SupportsFloat, SupportsInt, _SupportsRound):
+    # TODO: SupportsCeil, SupportsFloor, SupportsTrunc?
+
+    def __init__(cls, value: Union[_Decimal, float, Text,
```

Comment:
This is actually `__new__`, not `__init__`, at least for the Python implementation. It matters for subclassing.

Reply:
This is actually the whole reason I started to work on this. This problem was spotted by `find-problems.py` in https://github.com/srittau/typeshed-completion.